addons/phone_validation/models/models.py

```diff
@@ -20,6 +20,25 @@ def _phone_get_number_fields(self):
             number_fname for number_fname in ('mobile', 'phone') if number_fname in self
         ]
 
+    def _phone_get_country(self):
+        """Get a country likely to match the phone of the record.
+
+        By default we get it from:
+        - The country field of the target record (self)
+        - The country of any mail partner (e.g. self.partner_ids[2].phone), considering we are
+          going to contact the customer(s) of the record.
+        """
+        self.ensure_one()
+        record_country = None
```

Comment:
Would set as a void recordset, so that the returned type is always a record (you can always use result.id, which can be False or an id, easier to manipulate in callers).